src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -321,91 +326,89 @@ int MPIDI_OFI_get_huge_event(struct fi_cq_tagged_entry *wc, MPIR_Request * req)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_GET_HUGE_EVENT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_GET_HUGE_EVENT);
 
-    if (recv_elem->localreq && recv_elem->cur_offset != 0) {    /* If this is true, then the message has a posted
-                                                                 * receive already and we'll be able to find the
-                                                                 * struct describing the transfer. */
-        if (MPIDI_OFI_COMM(recv_elem->comm_ptr).enable_striping) {
-            /* Subtract one stripe_chunk_size because we send the first chunk via a regular message
-             * instead of the memory region */
-            recv_elem->stripe_size = (recv_elem->remote_info.msgsize - MPIDI_OFI_STRIPE_CHUNK_SIZE)
-                / MPIDI_OFI_global.num_nics;    /* striping */
-
-            if (recv_elem->stripe_size > MPIDI_OFI_global.max_msg_size) {
-                recv_elem->stripe_size = MPIDI_OFI_global.max_msg_size;
-            }
-            if (recv_elem->chunks_outstanding)
-                recv_elem->chunks_outstanding--;
-            bytesLeft = recv_elem->remote_info.msgsize - recv_elem->cur_offset;
-            bytesToGet = (bytesLeft <= recv_elem->stripe_size) ? bytesLeft : recv_elem->stripe_size;
-        } else {
-            /* Subtract one max_msg_size because we send the first chunk via a regular message
-             * instead of the memory region */
-            bytesLeft = recv_elem->remote_info.msgsize - recv_elem->cur_offset;
-            bytesToGet = (bytesLeft <= MPIDI_OFI_global.max_msg_size) ?
-                bytesLeft : MPIDI_OFI_global.max_msg_size;
-        }
-        if (bytesToGet == 0ULL && recv_elem->chunks_outstanding == 0) {
-            MPIDI_OFI_send_control_t ctrl;
-            /* recv_elem->localreq may be freed during done_fn.
-             * Need to backup the handle here for later use with MPIDIU_map_erase. */
-            uint64_t key_to_erase = recv_elem->localreq->handle;
-            recv_elem->wc.len = recv_elem->cur_offset;
-            recv_elem->done_fn(&recv_elem->wc, recv_elem->localreq, recv_elem->event_id);
-            ctrl.type = MPIDI_OFI_CTRL_HUGEACK;
-            mpi_errno =
-                MPIDI_OFI_do_control_send(&ctrl, NULL, 0, recv_elem->remote_info.origin_rank,
-                                          recv_elem->comm_ptr, recv_elem->remote_info.ackreq);
-            MPIR_ERR_CHECK(mpi_errno);
+    MPIR_Assert(recv_elem->localreq && recv_elem->cur_offset != 0);
```

Comment:
This assertion is still incomplete if you're trying to verify that the message has arrived. The check I added in #5357 is still needed.

Given that you're refactoring this code to make that PR obsolete, I'd say that you should dump the assertion here and add a description at the beginning that says that this function shouldn't be called unless the message has been both received and matched.

Even if you don't add that comment and want to leave this assertion, you should still have the comment. Otherwise it's not clear what this assertion is testing for.

Reply:
How about deleting the assertion altogether? I added it mostly for debugging purposes -- to catch cases that I wasn't aware of yet. Now the tests pass -- proving that I caught all the cases, the assertion should be deleted, especially since now seems it is distracting rather than clearing the code.

Reply:
I agree that the assertion is unnecessary, but I would still like to see more function descriptions and this seems like a good place to start.

Reply:
Ok, will add more descriptions

Reply:
Comments added https://github.com/pmodels/mpich/blob/74d914ac168d0f03b755b366283d63b8c178a13e/src/mpid/ch4/netmod/ofi/ofi_events.c#L329-L339